src/wp-includes/media.php

```diff
@@ -1736,7 +1736,7 @@ function wp_lazy_loading_enabled( $tag_name, $context ) {
 	// By default add to all 'img' and 'iframe' tags.
 	// See https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-loading
 	// See https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-loading
-	$default = ( 'img' === $tag_name || 'iframe' === $tag_name );
+	$default = false !== $context && ( 'img' === $tag_name || 'iframe' === $tag_name );
```

Comment:
There are several places we could adjust for block theme behavior, what do you think @felixarntz?

Reply:
@adamsilverstein When is the `$context` false here? Does the post content in a block theme not go through `the_content`?

Reply:
when a block theme is enabled, `the_content` filter is applied to the content with `$context` false which appears to happen before the main application of the filter. I'm going to dig a bit further to see why that is happening, maybe we can avoid it in the first place.

Reply:
Probably this:

https://github.com/WordPress/wordpress-develop/blob/874aa24b4e4437e14d336a8c3af32fa2039291ef/src/wp-includes/blocks/post-content.php#L55

Reply:
@adamsilverstein Hmm that code doesn't look odd to me. The `$context` is never explicitly specified in that filter, it uses `current_filter()` by default, which means here it should be set to `the_content`. This would only be a problem if some core code called `wp_filter_content_tags()` directly without the parameter (i.e. outside of a filter).

Reply:
indeed I verified that isn't the problem, I'm going to dig in more to trace back and find where the false context value is coming from.

Reply:
@felixarntz I dug in a bit further here...
`wp_filter_content_tags` is called **several times** by block themes in two places
:   in `src/wp-includes/blocks/template-part.php`::`render_block_core_template_part` with no context and `current_filter` returning false.  

Removing this call fixed the issue (but I don't think we can do that):

https://github.com/WordPress/wordpress-develop/blob/20bccea8a7b079be1f492ad3f59c9dc954c4ab01/src/wp-includes/blocks/template-part.php#L137

note: also called in `src/wp-includes/block-template.php` in `get_the_block_template_html`:

https://github.com/WordPress/wordpress-develop/blob/5eddc8f18c97d2eb909d0f70f4e20732398a7bda/src/wp-includes/block-template.php#L244

I considered some other fixes but the check in the current PR seemed the most reliable, what do you think?

Reply:
Sample trace:
```
wp_filter_content_tags (/wp-includes/media.php:1782)
render_block_core_template_part (/wp-includes/blocks/template-part.php:137)
WP_Block->render (/wp-includes/class-wp-block.php:256)
render_block (/wp-includes/blocks.php:1051)
do_blocks (/wp-includes/blocks.php:1089)
get_the_block_template_html (/wp-includes/block-template.php:240)
include (/wp-includes/template-canvas.php:12)
require_once (/wp-includes/template-loader.php:106)
require (/wp-blog-header.php:19)
{main} (/index.php:17)
```